tests/run.sh

```diff
@@ -17,21 +17,21 @@ if [ "$CI" = true ]; then
     set -x
 fi
 
+# Change to file's dir (POSIXly).
+cd -P -- "$(dirname -- "$0")"
+this_dir=$PWD
+source_dir="$PWD/.."
+
 # Either the build dir is passed in $CMAKE_BINARY_DIR or we guess based on $PWD
 build_dir="$CMAKE_BINARY_DIR"
 if [ -z "$build_dir" ]; then
-    if [ -d "$PWD/build" ]; then
-        build_dir="$PWD/build"
+    if [ -d "$source_dir/build" ]; then
+        build_dir="$source_dir/build"
     else
         build_dir="$PWD"
```

Comment:
Shouldn't this also be `build_dir="$source_dir"`, just like the change you did to the previous line?
(No idea which of these cases is run on travis, thinks are confusing and complicated...)